wafer/pages/models.py

```diff
@@ -22,7 +22,11 @@ class File(models.Model):
     item = models.FileField(upload_to='pages_files')
 
     def __str__(self):
-        return u'%s' % (self.name,)
+        if self.pages.exists():
```

Comment:
The major issue is the display of the files in the sponsor admin interface, which uses a simple list. It isn't trivial to replace the use of str there with something else.

We could reduce this to just using the name and url, which should allow distinguishing files well enough.